saq/queue/postgres.py

```diff
@@ -0,0 +1,334 @@
+"""
+Postgres Queue
+"""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import time
+import typing as t
+
+from psycopg.types.json import Jsonb
+from psycopg_pool import AsyncNullConnectionPool, AsyncConnectionPool
+
+from saq.job import (
+    Job,
+    Status,
+)
+from saq.queue.base import Queue, logger
+from saq.utils import now
+
+if t.TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from saq.types import (
+        CountKind,
+        ListenCallback,
+        DumpType,
+        LoadType,
+        QueueInfo,
+        QueueStats,
+    )
+
+JOBS_TABLE = "saq_jobs"
+
+CREATE_JOBS_TABLE = """
+CREATE TABLE IF NOT EXISTS saq_jobs (
+    key TEXT PRIMARY KEY,
```

Comment:
we'll need some other indices, like something that partitions on status, scheduled, etc